lib/src/main/java/com/dynatrace/metric/util/MetricLineBuilderImpl.java

```diff
@@ -81,7 +94,13 @@ public MetricLineBuilder.TypeStep metricKey(String key) throws MetricException {
       NormalizationResult normalizationResult = Normalizer.normalizeMetricKey(this.metricKey);
 
       if (normalizationResult.messageType() == NormalizationResult.MessageType.WARNING) {
-        logger.warning(() -> normalizationResult.getMessage());
+        if (!metricKeyNormalizationWarnLogged) {
```

Comment:
We could consider wrapping these booleans into a class, and then have specific methods for each case there. So here it would be simpler, something like `NormalizationThrottledLogger.logMetricKeyNormalization()`